source/fab/reader.py

```diff
@@ -5,9 +5,10 @@
 
 
 class TextReader(ABC):
+    @property
     @abstractmethod
```

Comment:
Weirdly enough... when I did this one of the checkers complained, and I ended up having to put `@abstractproperty` (imported from ABC)... I think perhaps I had `abstractmethod` and `property` the other way around to this.... either way maybe the above is neater?

Reply:
According to the documentation `@abstractproperty` was deprecated in Python 3.3.

Reply:
Oh!  In that case I have some updates to make in my other PR